solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -40,10 +40,38 @@ fn get_escrow_account(
     denom: &str,
 ) -> Pubkey {
     let denom = lib::hash::CryptoHash::digest(denom.as_bytes());
-    let seeds = [port_id.as_bytes(), channel_id.as_bytes(), denom.as_slice()];
+    let seeds = [
+        crate::ESCROW,
+        port_id.as_bytes(),
+        channel_id.as_bytes(),
+        denom.as_slice(),
+    ];
     Pubkey::find_program_address(&seeds, &crate::ID).0
 }
 
+fn get_token_mint(denom: &PrefixedDenom) -> Result<Pubkey, TokenTransferError> {
+    let denom = denom.to_string();
+    let denom: Vec<&str> = denom.split('/').collect();
```

Comment:
By the way, Split is a DoubleEndedIterator so this could theoretically be done without collecting to a vector though getting the length for the error isn’t pretty:

    let denom = denom.base_denom.as_str().as_bytes();
    let denom = CryptoHash::digest(denom);

    let path = denom.trace_path.to_string();
    let mut path = path.split('/');
    let channel_id = path.next_back();  // Channel id is last
    let port_id = path.next_back();     // Port id is second to last
    let (channel_id, port_id) = match (port_id, channel_id) {
        (Some(port_id), Some(channel_id)) => (port_id, channel_id),
        (_, last) => return Err(TokenTransferError::InvalidTraceLength {
            len: path.count() as u64 + u64::from(last.is_some()),
        },
    };

Dunno if it’s worth the hassle though.

Reply:
i dont mind going with this.